components/telemetry-operator/controller/logpipeline/controller.go

```diff
@@ -44,10 +44,11 @@ type Config struct {
 // Reconciler reconciles a LogPipeline object
 type Reconciler struct {
 	client.Client
-	config           Config
-	syncer           *syncer
-	unsupportedTotal prometheus.Gauge
-	daemonSetHelper  *kubernetes.DaemonSetHelper
+	config                  Config
+	syncer                  *syncer
+	daemonSetHelper         *kubernetes.DaemonSetHelper
+	allLogPipelines         prometheus.Gauge
+	unsupportedLogPipelines prometheus.Gauge
```

Comment:
Just an idea: What do you think about (similar to the `syncer`) wrap metrics into a `metrics` or `controllerMetrics` struct, where the constructor function create the Gauges and possibly future metrics can be added.

Also calls to this would look like `metrics.Update(&allPipelines)`

Reply:
I agree that it would make sense to collect all metrics under an overarching struct. However, having a single point of the metric update is hard because different metrics are updated in different phases of reconciliation.